sycl/source/detail/graph/graph_impl.cpp

```diff
@@ -740,14 +733,12 @@ void exec_graph_impl::findRealDeps(
   }
 }
 
-ur_exp_command_buffer_sync_point_t
-exec_graph_impl::enqueueNodeDirect(const sycl::context &Ctx,
-                                   sycl::detail::device_impl &DeviceImpl,
-                                   ur_exp_command_buffer_handle_t CommandBuffer,
-                                   std::shared_ptr<node_impl> Node) {
+ur_exp_command_buffer_sync_point_t exec_graph_impl::enqueueNodeDirect(
```

Comment:
This and `enqueue` are called at around line 870, with `shared_ptr`s removed from that area of the code. I could have used `shared_from_this()` and kept these two the same, but they don't seem do be called anywhere else and it made sense to me to update them as part of this PR.